### PR TITLE
switch to using libinput exclusively (x86 / Generic only)

### PR DESCRIPTION
**Pull request by lrusak**
Let's give this a go again.

positive reports so far from @MilhouseVH

### CODE REVIEW

File: packages/virtual/x11/package.mk [hunk 1]
```diff
@@ -49,11 +49,7 @@ fi
 get_graphicdrivers
 
 # Drivers 
-if [ -n "$LIBINPUT" ]; then
   PKG_DEPENDS_TARGET="$PKG_DEPENDS_TARGET xf86-input-libinput"
```
Comment (MindTooth) [inline packages/virtual/x11/package.mk]: Indentation correct?

**Comment by lrusak** [comment]
@MilhouseVH any feedback here?

**Comment by MilhouseVH** [comment]
There's been a couple of issues reported:

http://forum.kodi.tv/showthread.php?tid=269815&pid=2318869#pid2318869
http://forum.kodi.tv/showthread.php?tid=269815&pid=2322219#pid2322219 (later http://forum.kodi.tv/showthread.php?tid=269815&pid=2322663#pid2322663)

**Comment by lrusak** [comment]
Maybe @whot would be so kind to help us out with out remote issues.

Maybe it's just a matter of shipping both xf86-input-evdev and xf86-input-libinput together? I think libinput rules would override the former but I'm not 100% sure.

**Comment by whot** [comment]
how do you guys use the remote? can you file a bug against libinput on bugs.freedesktop.org and attach the evemu-describe output from one of these devices?

libinput does not really support remotes yet, it probably handles (some of) them as keyboards but that's not the long-term plan. So in the meantime you should keep using evdev against remotes, but that would require custom xorg.conf snippets since libinput by default overrides evdev for keyboards (and remotes largely look like keyboards).

**Comment by MilhouseVH** [comment]
There have been several reports in the test thread of auto-repeat not working since the switch to libinput. I'm going to drop this PR from the next test build - let's see if all the remote issues clear up. Assuming this PR is responsible for these issues, this PR should be put on hold or closed until there's a solution.

**Comment by lrusak** [comment]
@whot The remotes are typically used through lirc or eventlircd. I'll see what I can do about getting a bug report going, however, I don't have any of these remotes so we'll have to get a user to volunteer.

**Comment by MilhouseVH** [comment]
Dropping this PR has restored the "missing buttons" but hasn't fixed the auto-repeat issue, could this be due to the kernel change (the switch to libinput and 4.6-rc4 occurred at the same time, build #0423). I'm going to do a clean build for tonight just to be absolutely sure...

**Comment by whot** [comment]
fwiw, libinput does nothing with auto-repeat, like evdev it swallows kernel repeat events and relies on the higher levels to do repeating.

**Comment by chewitt** [comment]
@lrusak is this still relevant?

**Comment by lrusak** [comment]
I use libinput in all my generic builds.

It would be nice to switch but we need to find a good way to test remotes so that they still work properly.

**Comment by chewitt** [comment]
@lrusak sending a bi-annual ping on this. It's not part of the OS that I understand :)

**Comment by lrusak** [comment]
It would be worth a test again. We need to nail down which remotes break when libinput is used.

**Comment by chewitt** [comment]
@lrusak given the general mid-term direction towards exorcising Xorg can this be closed?

**Pull request closed without merging**